src/da/cmd-handler.js

```diff
@@ -0,0 +1,176 @@
+ */
+import fs from 'fs';
+import chalk from 'chalk';
+import fetch from 'node-fetch';
+import { uploadFolder } from './upload.js';
+import { processAndUpdateHTMLPages } from './da-helper.js';
+
+// DA API base URL
+const DA_BASE_URL = 'https://admin.da.live';
+
+/**
+ * Validate the existence of the asset-mapping.json and HTML folder.
+ * @param {string} assetMappingFile - The path to the asset-mapping.json file
+ * @param {string} htmlFolder - The path to the HTML folder
+ * @return {boolean} True if the files exist, false otherwise
+ */
+function validateFiles(assetMappingFile, htmlFolder) {
+  const files = [
+    { path: assetMappingFile, message: `asset-mapping.json file not found: ${assetMappingFile}`, mandatory: true },
+    { path: htmlFolder, message: `HTML folder not found: ${htmlFolder}`, mandatory: true },
+  ];
+
+  for (const file of files) {
+    if (file.mandatory === false) {
+      continue;
+    }
+
+    if (!fs.existsSync(file.path)) {
+      console.error(chalk.red(file.message));
+      return false;
+    }
+  }
+  return true;
+}
+
+/**
+ * Validate the DA authentication token by making a HEAD request to the target environment.
+ * @param {string} daLocation - The DA target environment
+ * @param {string} token - The DA authentication token
+ * @return {Promise<boolean>} True if the token is valid, false otherwise
+ */
+async function validateLogin(listUrl, token) {
+  try {
+    
+    const headers = {
+      Authorization: `Bearer ${token}`,
+    };
+    const response = await fetch(listUrl, { method: 'HEAD', headers });
+
+    if (!response.ok) {
+      console.error(chalk.red(`Login failed with status: ${response.status} - ${response.statusText}`));
+      if (response.status === 401 || response.status === 403) {
+        console.error(chalk.red('Unauthorized: Invalid token'));
+      }
+      return false;
+    }
+
+    const text = await response.text();
+    if (text.includes('Invalid token') || text.includes('Unauthorized')) {
+      console.error(chalk.red(`Invalid token detected in response body: ${text}`));
+      return false;
+    }
+
+    return response.status === 200;
+  } catch (error) {
+    console.error(chalk.red(`Network error: ${error.message}`));
+    return false;
+  }
+}
+
+export const daBuilder = (yargs) => {
+  return yargs
+    .option('org', {
+      type: 'string',
+      describe: 'The organization',
+      demandOption: true,
+    })
+    .option('repo', {
+      type: 'string',
+      describe: 'Name of the repository',
+      demandOption: true,
+    })
+    .option('asset-url-list', {
+      type: 'string',
+      describe: 'Absolute path to the asset-url.json file',
+      demandOption: true,
+    })
+    .option('da-folder', {
+      type: 'string',
+      describe: 'Absolute path to the DA folder',
+      demandOption: true,
+    })
+    .option('download-folder', {
+      type: 'string',
+      describe: 'Path to the download folder',
+      demandOption: true,
```

Comment:
```suggestion
      default: 'da-assets',
```

Can we make this optional, and if not provided use the default fallback?